packages/scene-composer/src/components/panels/CustomColorPicker/CustomColorPicker.tsx

```diff
@@ -0,0 +1,168 @@
+  );
+  const [hexInputValue, setHexInputValue] = useState<string>(() =>
+    !!color && isValidHex(color) ? color : DEFAULT_COLOR_PICKER_COLOR.hex,
+  );
+  const [rgbInputValue, setRgbInputValue] = useState<RGB>(() => {
+    return !!color && isValidHex(color) ? hexToRgb(color) : DEFAULT_COLOR_PICKER_COLOR.rgb;
+  });
+
+  const handleColorfulChange = (newColor: string) => {
+    setInternalColor(newColor);
+    setHexInputValue(newColor);
+    setRgbInputValue(hexToRgb(newColor));
+  };
+
+  const handleHexChange = (newColor: string) => {
+    setHexInputValue(validateHex(newColor));
+    if (isValidHex(newColor)) {
+      setInternalColor(newColor);
+      setRgbInputValue(hexToRgb(newColor));
+    }
+  };
+
+  const handleRGBChange = (newValue: Partial<RGB>) => {
+    setRgbInputValue({ ...rgbInputValue, ...newValue });
+    setHexInputValue(RgbToHex({ ...rgbInputValue, ...newValue }));
+    setInternalColor(RgbToHex({ ...rgbInputValue, ...newValue }));
+  };
+
+  const onSubmitAttempt = () => {
+    setSaveAttempted(true);
+    //The hex input is the only field that could conceivably be invalid
+    const valid = isValidHex(hexInputValue);
+    if (valid) {
+      onSubmit(internalColor);
+    } else {
+      hexInputFieldRef.current?.focus();
+    }
+  };
+
+  return (
+    <ColorPickerWrapper>
+      <form onSubmit={(e) => e.preventDefault()}>
+        <div className='color-picker' data-testid='react-colorful-picker'>
+          <HexColorPicker color={internalColor} onChange={debounce(handleColorfulChange, 100)} />
+        </div>
+
+        <SpaceBetween size='s'>
+          <FlexibleDiv>
+            <FormField
+              label={formatMessage({ defaultMessage: 'Hex', description: 'Abbreviation for hexidecimal' })}
+              errorText={
+                !isValidHex(hexInputValue) &&
+                saveAttempted &&
+                formatMessage({ defaultMessage: 'Invalid Hex', description: 'Form field error message' })
+              }
+              data-testid='hex-input-form-field'
+            >
+              <div role='status'>
+                <FixedWidthInput
+                  autoFocus={!!onCancel}
+                  width='75px'
+                  value={`#${hexInputValue.replace('#', '')}`}
+                  onChange={(e) => handleHexChange(e.detail.value)}
+                  ref={hexInputFieldRef}
+                  data-testid='hex-input'
+                />
+              </div>
+            </FormField>
+            <SpaceBetween size='xs' direction='horizontal'>
+              <FormField label={formatMessage({ defaultMessage: 'R', description: 'Abbreviation for red' })}>
+                <div role='status'>
+                  <FixedWidthInput
+                    width='42px'
+                    value={rgbInputValue.r}
+                    onChange={(e) => handleRGBChange({ r: validateRgbValue(e.detail.value) })}
+                    data-testid='red-input'
```

Comment:
Would using a number input here work? Polaris implementation [here](https://cloudscape.aws.dev/components/input/?tabId=playground&props=N4IgbghgNgrgpiAXCAEnKUD2ACA6pgJygBMQAaECGAF0wGFMBbAByjmoWWoPnJAEsAdsxoBZTMU4hBMRnAL8Axn2oBPZlJmMARvL4BnDsyQgQAXyA). You wouldn't have to worry about validating `onlyDigits` and I think there are some benefits to browsers knowing the type of input expected

Reply:
Good point, will change